pkg/tfbridge/x/token_test.go

```diff
@@ -169,6 +169,112 @@ func TestIgnored(t *testing.T) {
 	}, info.Resources)
 }
 
+func TestTokensInferredModules(t *testing.T) {
+	tests := []struct {
+		name            string
+		resourceMapping map[string]string
+		opts            *InferredModulesOpts
+	}{
+		{
```

Comment:
Suggest adding example here based on our conversation from OCI with a bit of long-form explanation motivating the tree algorithm tricks.

```

			"oci_apm_apm_domain": {Tok: tfbridge.MakeResource(mainPkg, apmMod, "ApmDomain")},

			"oci_apm_config_config": {Tok: tfbridge.MakeResource(mainPkg, apmConfigMod, "Config")},

			"oci_apm_synthetics_monitor":                 {Tok: tfbridge.MakeResource(mainPkg, apmSyntheticsMod, "Config")},
			"oci_apm_synthetics_script":                  {Tok: tfbridge.MakeResource(mainPkg, apmSyntheticsMod, "Script")},
			"oci_apm_synthetics_dedicated_vantage_point": {Tok: tfbridge.MakeResource(mainPkg, apmSyntheticsMod, "DedicatedVantagePoint")},

```

The algo looking at "oci_apm_apm_domain" needs to figure out if the module is oci_apm or oci_apm_apm. It's not allowed to know the list of all known submodules. So the way it does inference is it notices that there's only one "oci_apm_apm" prefix, therefore it's not a module; if the user added "oci_apm_apm_thingy" resource, then there'd be two entries and it might decide oci_apm_apm is now a module.